src/components/Meals.tsx

```diff
@@ -1,66 +1,65 @@
 import React, { useState } from 'react';
-import { View, Text, TextInput, TouchableOpacity, Button } from 'react-native';
+import { View, Text, TextInput, TouchableOpacity } from 'react-native';
 import styles from '../styles';
 import { MealCategories } from '../models/meals.model';
 
 type MealsComponentProps = {
   mealCategories: MealCategories[];
-  handleBreakfastText: (text: string, category: MealCategories) => void;
+  handleMealText: (text: MealText, category: MealCategories) => void;
 };
 
-const Meals: React.FC<MealsComponentProps> = ({ mealCategories, handleBreakfastText }) => {
-  const [textMeal, setMealText] = useState('');
-  const [breakfastText, setBreakfastText] = useState('');
-  const [lunchText, setLunchText] = useState('');
-  const [dinnerText, setDinnerText] = useState('');
+type MealText = string;
 
-  const onMealEnter = (text: string, category: MealCategories) => {
-    switch (category) {
-      case MealCategories.BREAKFAST:
-        setBreakfastText(text);
-        break;
-      case MealCategories.LUNCH:
-        setLunchText(text);
-        break;
-      case MealCategories.DINNER:
-        setDinnerText(text);
-        break;
-      default:
-        break;
-    }
+const Meals: React.FC<MealsComponentProps> = ({ mealCategories, handleMealText }) => {
+  const [mealText, setMealText] = useState<Record<MealCategories, MealText>>({
+    [MealCategories.BREAKFAST]: '',
+    [MealCategories.LUNCH]: '',
+    [MealCategories.DINNER]: '',
+  });
+  const [showEnteredText, setShowEnteredText] = useState<Record<MealCategories, boolean>>({
+    [MealCategories.BREAKFAST]: false,
+    [MealCategories.LUNCH]: false,
+    [MealCategories.DINNER]: false,
+  });
+
+  const onMealEnter = (text: MealText, category: MealCategories) => {
+    setMealText((prev) => ({
+      ...prev,
+      [category]: text,
+    }));
   };
 
   const getCategoryInput = (category: MealCategories) => {
```

Comment:
not sure u need a function for this, but if yes then u can shorten it like this if u want
```suggestion
  const getCategoryInput = (category: MealCategories) =>  mealText[category];
```

Reply:
ya i got rid of the function